client/src/main/java/io/pravega/client/stream/impl/EventStreamReaderImpl.java

```diff
@@ -59,7 +61,7 @@
 public class EventStreamReaderImpl<Type> implements EventStreamReader<Type> {
 
     // Base waiting time for a reader on an idle segment waiting for new data to be read.
-    private static final long BASE_READER_WAITING_TIME_MS = 10;
+    private static final long BASE_READER_WAITING_TIME_MS = ReaderGroupStateManager.TIME_UNIT.toMillis();
```

Comment:
The waiting time is now increased from 10ms to 1000ms .Any specific reason for this ?

Reply:
Originally it was 1000ms and we used futures to wake up the blocked thread. However futures created the issue linked above, so we changed it to 10ms so that while it would not wakeup in response to data being available, it least it would check every 10ms. Now that we have a wakeup again I am restoring it to 1000ms. This is fine because it will awake in response to data. We still need to wake up eventually though, the source of the timeout shows why, even if there is no data, there might be an update to the group state, which we pull in updates for every 1000ms.